### ISSUE TITLE
index out of range runtime error in slash_dmy.go

### ISSUE THREAD

**Issue by cpalsulich** [olebedev/when#36]
The `getDays` method attempts to access array elements using a [passed in month argument as an index](https://github.com/olebedev/when/blob/master/rules/common/slash_dmy.go#L32). The problem is month can be larger than 12 because [the regex](https://github.com/olebedev/when/blob/master/rules/common/slash_dmy.go#L38C2-L44) defined in `SlashDMY` matches on dates in a mm/dd/yyyy format. So this means a date like `1/30/2023` will cause a panic.

**Comment by cpalsulich** [comment]
Relevant stack trace lines:
```
runtime error: index out of range [21] with length 13
  /go/pkg/mod/github.com/olebedev/when@v0.0.0-20221205223600-4d190b02b8d8/rules/common/slash_dmy.go:32
  /go/pkg/mod/github.com/olebedev/when@v0.0.0-20221205223600-4d190b02b8d8/rules/rules.go:42
```

**Comment by olebedev** [comment]
Hello @cpalsulich, thank you for your report. There are two separate issues with the library:

1. The library is currently causing panics, which is an unexpected behavior. We have to treat this as a bug or missing feature and resolve it accordingly. Pull requests are welcome!

1. Regarding the parsing of the US date standard (`mm/dd/yyyy`) using the default set of rules defined in [this](https://github.com/olebedev/when/blob/master/when.go#L159) line, it is unclear whether this is a bug or intended behavior. The rule implemented in `slash_dmy.go` is specifically designed to parse the `dd/mm/yyyy` format indicated by its file name. Therefore, expecting successful parsing of the `mm/dd/yyyy` format is not appropriate in this case.

To resolve this issue, you will need to create a custom rule that handles the desired date format. Pull requests are very welcome, as always!

Hope that helps,
Oleg

**Comment by olebedev** [comment]
Related #10

**Comment by olebedev** [comment]
Resolved in #37
